kfp_tests/run_integration_tests.py

```diff
@@ -33,46 +34,60 @@
 
 
 def parse_run_id(output):
-    run_id_flag = False
-    run_id = ""
-    for i, char in enumerate(output):
-        if i >= 7 and output[i - 7 : i] == "run_id:":
-            run_id_flag = True
-        if run_id_flag and char == "\n":
-            return run_id
-        if run_id_flag:
-            run_id += char
-    return -1
+    if "run_id|" not in output or "|end_id" not in output:
+        return -1
+    start = output.find("run_id|") + len("run_id|")
+    end = output.find("|end_id")
+    run_id = output[start:end]
+    return run_id
 
 
 def obtain_flow_file_paths(flow_dir_path):
     file_paths = [f for f in listdir(flow_dir_path) if isfile(join(flow_dir_path, f))]
-    file_paths.remove("__init__.py")  # __init__.py needed to establish Python module
     return file_paths
 
 
+@ray.remote
+def spawn_and_return(args, flow_file_path):
+    """
+    This function is the creation and waiting one KFP run. Since
+    we don't want to wait for the entire flow to finish on KFP before
+    creating a new run, we will have Ray parallelize the effort.
+    """
+    full_path = join(args.flow_dir_path, flow_file_path)
+    process = run(
+        ["python", full_path, "--datastore=s3", "kfp", "run"],
+        stdout=PIPE,
+        stderr=PIPE,
+        check=True,
+        text=True,
+    )
+    run_id = parse_run_id(process.stdout)
+    if run_id != -1:
+        run_link = f"{KFP_RUN_URL_PREFIX}/_/pipeline/#/runs/details/{run_id}"
+        print(f"Running {flow_file_path} ... Run link: {run_link}")
+        kfp_client = kfp.Client(namespace=args.namespace, userid=args.userid)
+        run_response = kfp_client.wait_for_run_completion(run_id, 500).to_dict()
+        if run_response["run"]["status"] == "Succeeded":
+            print(f"Flow {flow_file_path} ran successfully! Run link: {run_link}")
+        else:
+            print(
+                f"Flow {flow_file_path} encountered a runtime error. Run link: {run_link}"
+            )
+    else:
+        print(f"Flow {flow_file_path} failed to compile properly.")
+    return
```

Comment:
We can return something or throw error to indicate if a test fail.

Reply:
Let me investigate what happens if we throw an exception inside a process that is running remotely with Ray. For now, I agree.

Reply:
@cloudw according to Taleb's suggestion, we print if a pipeline fails, and log it to the screen. No need for exceptions for run time issues.

Reply:
Ok this also works for now. We can revisit later if needed.